src/MediaPresenter.ts

```diff
@@ -0,0 +1,157 @@
+import * as mobx from "mobx";
+
+const MAXLEN = 10;
+
+export class MediaStore {
+  file: File;
+  duration: number = 0;
+  played: number = 0;
+  constructor(newFile: File) {
+    this.file = newFile;
+    // duration of each media attached
+    this.duration = 0;
+    // played list shows how long each of the file have been played
+    this.played = 0;
+  }
+}
+
+export class MediaPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+
+  @mobx.observable.deep
+  media: MediaStore[] = [];
+
+  static isImage(file: File): boolean {
+    const imageFormat = new RegExp("image/*");
+    return imageFormat.test(file.type);
+  }
+
+  static isVideo(file: File): boolean {
+    const videoFormat = new RegExp("video/*");
+    return videoFormat.test(file.type);
+  }
+
+  @mobx.action
+  addFile(newFile: File): boolean {
+    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
+      this.media.push(new MediaStore(newFile));
+      return true;
+    }
+    return false;
+  }
+
+  @mobx.action
+  removeFile(fileIndex: number): void {
+    if (this.media.length === 0) {
+      return;
+    }
+    let index = fileIndex % this.filesLength;
+    if (index > -1) {
+      this.media.splice(index, 1);
+    }
+  }
+
+  @mobx.action
+  setDuration(fileIndex: number, duration: number): void {
+    // make sure that accessing files is % filesLength,
+    // so that it loops back to the start of the file if the video is too long
+    const modIndex = fileIndex % this.filesLength;
+    this.media[modIndex].duration = duration;
```

Comment:
As a further tidy up: you're calling `this.media[fileIndex % this.filesLength]` in a lot of different fns (here, getDuration, getFileName, get File etc.)
Instead, you could create a `getMedia` fn, as follows:

```
   getMedia(fileIndex: number): MediaStore {
       // make sure that accessing files is % filesLength,
       // so that it loops back to the start of the file if the video is too long
       const modIndex = fileIndex % this.filesLength;
       return this.media[modIndex];
   }
```

Then you can just do `this.getMedia(fileIndex).duration` (or whatever property you need), and the mod check is contained in one place